packages/amplify-category-auth/src/provider-utils/awscloudformation/import/index.ts

```diff
@@ -0,0 +1,1286 @@
+  // We can't validate until fully until AppClients are selected later.
+  if (answers.authSelections === 'identityPoolAndUserPool') {
+    const identityPools = await identity.listIdentityPoolDetails();
+
+    const identityPoolCandidates = identityPools
+      .filter(
+        ip => ip.CognitoIdentityProviders && ip.CognitoIdentityProviders!.filter(a => a.ProviderName?.endsWith(userPoolId)).length > 0,
+      )
+      .map(ip => ({
+        identityPool: ip,
+        providers: ip.CognitoIdentityProviders!.filter(a => a.ProviderName?.endsWith(userPoolId)),
+      }));
+
+    var validatedIdentityPools: { identityPool: IdentityPool; providers: CognitoIdentityProvider[] }[] = [];
+
+    for (const candidate of identityPoolCandidates) {
+      const hasWebClientProvider =
+        candidate.providers.filter(p => p.ClientId && webClients.map(c => c.ClientId).includes(p.ClientId!)).length > 0;
+      const hasNativeClientProvider =
+        candidate.providers.filter(p => p.ClientId && nativeClients.map(c => c.ClientId).includes(p.ClientId!)).length > 0;
+
+      if (hasWebClientProvider && hasNativeClientProvider) {
+        validatedIdentityPools.push(candidate);
+      }
+    }
+
+    if (validatedIdentityPools.length === 0) {
+      return importMessages.NoIdentityPoolsFoundWithSelectedUserPool;
+    }
+
+    parameters.validatedIdentityPools = validatedIdentityPools;
+  }
+
+  // Save into parameters, further quesions are using it
+  if (parameters.webClients?.length === 0) {
+    parameters.webClients!.push(...(webClients || []));
+  }
+  if (parameters.nativeClients?.length === 0) {
+    parameters.nativeClients!.push(...(nativeClients || []));
+  }
+
+  return true;
+};
+
+const selectAppClients = async (
+  context: $TSContext,
+  enquirer: Enquirer<ImportAnswers>,
+  questionParameters: ImportParameters,
+  answers: ImportAnswers,
+): Promise<void> => {
+  let autoSelected = 0;
+
+  // Select web application clients
+  if (questionParameters.webClients!.length === 1) {
+    answers.appClientWeb = questionParameters.webClients![0];
+
+    context.print.info(importMessages.SingleAppClientSelected('Web', answers.appClientWeb.ClientName!));
+
+    autoSelected++;
+  } else {
+    const appClientChoices = questionParameters
+      .webClients!.map(c => ({
+        message: `${c.ClientName!} (${c.ClientId})`,
+        value: c.ClientId,
+      }))
+      .sort((a, b) => a.message.localeCompare(b.message));
+
+    const appClientSelectQuestion = {
+      type: 'select',
+      name: 'appClientWebId',
+      message: importMessages.Questions.SelectAppClient('Web'),
+      required: true,
+      choices: appClientChoices,
+    };
+
+    context.print.info(importMessages.MultipleAppClients('Web'));
+
+    const { appClientWebId } = await enquirer.prompt(appClientSelectQuestion);
+    answers.appClientWeb = questionParameters.webClients!.find(c => c.ClientId! === appClientWebId);
```

Comment:
Answers are being mutated, mutating object parameter properties is fine, what is not fine is to reassign the incoming parameter. Since these params are passed by ref, we only maintaining 1 single answer object.